src/Planner/PlannerContext.cpp

```diff
@@ -106,9 +107,14 @@ TableExpressionData * PlannerContext::getTableExpressionDataOrNull(const QueryTr
 
 const ColumnIdentifier & PlannerContext::getColumnNodeIdentifierOrThrow(const QueryTreeNodePtr & column_node) const
 {
-    auto & column_node_typed = column_node->as<ColumnNode &>();
-    const auto & column_name = column_node_typed.getColumnName();
-    auto column_source = column_node_typed.getColumnSource();
+    auto * column_node_typed = column_node->as<ColumnNode>();
+    if (!column_node_typed)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS,
```

Comment:
Function name `getColumnNodeIdentifierOrThrow` and its documentation assume general purpose, which is inconsistent with the error message. Event though it's used only for join currently, consider renaming it then or moving check to caller side

Reply:
Agree. I'd prefer to see a changed exception message here